### DISCUSSION TITLE
Thank you nvidia!

### DISCUSSION THREAD

**Discussion by LinusCDE** [NVIDIA/open-gpu-kernel-modules#248] [General]
Hi,

just wanted to say thank you for doing this. I mean that honestly.

Whatever the business reason may be or whatever recent news story might have triggered this, I don't care.

I'm happily using the nvidia driver for at least 5 years on Linux now, and am quite happy. To be frank, then I first got my linux pc, the driver caused a lot of technical issues, I could barely understand and solve at that time. But since then I've been using the driver and was totally happy with it.

I also don't care that much about wayland rn. I know there is no decent alternative to this driver and wayland is kinda held back by it, but I'm not using wayland and therefore frankly just don't care rn. X is still fine for me and I'm also not that seldomly using the networking part that wayland ditched.

I also understand that this driver is not a "big chunk" of the still closed source being used by almost everyone who owns a nvidia gpu on linux. Opening more up could probably help for example:
 - Speeding up wayland integration which would probably move wayland as a whole forward
 - Updating the ancient looking user-space tools like the settings ui from the 90s or creating modern tools with as much features and integrations as the original one
 - Maybe some other stuff I don't realize (vr??)

Anyway I find this a good start and hope the community stops memeing here and sticks to at least well intended changes and not taking all the engineering time by responding to closed duplicates or responding nice to ill intended requests.

While I hope as stated above, that more code may be opened over the time. I already regard this as a great start and seriously want to thank you for taking these step. Whatever the true intention, the outcome is what matters. And over long time, I hope you can accept open source in this space more as the open source community also accepts you more.

Stuff takes time, problems like patents (e.g. hdmi stuff) and others issues can probably delay stuff a lot, but I hope - step by step - this you can start to make an impact to the open source community with opening up to feedback / improvements for your gtx and rtx gpus as well.

TL;DR: Just thank you. Whatever the reason or cause and however long it takes, I don't care. Only the result and intended future actions matter to me!

**Comment by aaronp24** [comment]
Thanks for the kind words, LinusCDE.

I just wanted to clarify one thing: the [settings UI](https://github.com/NVIDIA/nvidia-settings) has been open source for eighteen years. The first release, [1.0-6106](https://download.nvidia.com/XFree86/nvidia-settings/nvidia-settings-1.0-6106.tar.gz) has archive timestamps of Jun 23, 2004.

**Reply by LinusCDE** [reply]
Wow. Didn't know that one. Just assumed that the fact that "userspace tools" are not included meant the settings as well. That is pretty weird indeed that nobody ever wanted to change a thing.

Though, taking a quick at the prs, seems that some were done, but a good chunk never got done or properly discussed. So probably a bit of a two sided thing imo. Maybe the new attitude can change things there, too.
